src/ScriptProcessor.php

```diff
@@ -126,9 +127,17 @@ private function getTemplateAdapter(string $type): TemplateAdapter
         }
     }
 
-    public function processScript(string $dataDir, string $type, string $script): void
+    public function processScript(string $dataDir, string $type, ?string $script, ?array $codeChunks = null): void
```

Comment:
Where is it called with the `$codeChunks` parameter? (I don't see it here: https://github.com/keboola/data-loader/blob/941991f7c1d92926aa54fb1ff1713dca4b4ee999/src/run.php#L35)

Reply:
omg!  you're right, it's only called in the tests 🤦

Reply:
I started to look to it because it seems to me that these two if-else blocks have very similar then and else parts. The second on row 155 could be omitted if $script and $codeChunks are read as one parameter into this method. But then the first if block would not work, right? 🤔  I guess it's ok, just a note that I was stuck here for a moment. 😄

Reply:
Yeah, I don't really like how it's done here, but I plan on fixing it in the api.